### PR TITLE
[PP] Forward only schedule

### PR DESCRIPTION
**Pull request by H-Huang**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #132177

`python test/distributed/pipelining/test_schedule_multiproc.py -k test_forward_only`

cc @XilunWu @awgu @kwen2501 @wanchaol @fegin @fduwjj @wz337 @wconstab @d4l3k @c-p-i-o

### CODE REVIEW

File: test/distributed/pipelining/test_schedule_multiproc.py [hunk 1]
```diff
@@ -56,6 +57,47 @@ def setUpClass(cls):
...
+            self.rank,
+            self.device,
+        )
+
+        # Attach to a schedule
+        schedule = ScheduleClass(stage, num_microbatches)
+
+        # Run
+        num_iters = 20
+        for _ in range(num_iters):
```
Comment (lessw2020) [inline test/distributed/pipelining/test_schedule_multiproc.py:90]: generic q - is there an easy way to verify that things looped successfully (i.e. seed value for rand and then confirm expected final values after 20 iters)?
right now the test confirms it ran w/o error, but maybe future test should also verify an expected output?

**Review by lessw2020 — APPROVED** [review]
looks great, thanks a ton for adding! 
- minor q on future test addition
- linting errors are easy fixes

File: test/distributed/pipelining/test_schedule_multiproc.py (hunk 1 above)
Comment (H-Huang) [reply test/distributed/pipelining/test_schedule_multiproc.py:93]: Thanks! Added a validation check at the end of the test to compare the pipelined output with a reference model.

**Pull request closed without merging**